### PR TITLE
Launcher refactor

### PR DESCRIPTION
**Pull request by akashgupta9990**
Issue Link - https://project-sunbird.atlassian.net/browse/SB-5809?oldIssueView=true
Added 
1. Event "content:load:'content mimetype'" on all launcher.
2. Calling event "renderer:launcher:clean" before dispatching event to start content.
3. Removed destroy function to delete launcher from plugin manager.

### CODE REVIEW

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ org.ekstep.contentrenderer.baseLauncher.extend({
     stageId:[],
     qid:[],
     enableHeartBeatEvent:false,
+    sleep: false,
```
Comment (pallakartheekreddy) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js]: Change variable name to sleepMode

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js [hunk 2]
```diff
@@ -27,16 +28,19 @@ org.ekstep.contentrenderer.baseLauncher.extend({
     initLauncher: function(manifest) {
         EkstepRendererAPI.addEventListener('renderer:content:load', this.start, this);
         EkstepRendererAPI.addEventListener('renderer:cleanUp', this.cleanUp, this);
-        var instance = this;
-        setTimeout(function(){
-            instance.start();
-        }, 0);
+        EkstepRendererAPI.addEventListener('content:load:application/vnd.ekstep.ecml-archive', this.start, this);
```
Comment (pallakartheekreddy) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js]: Change event name as well

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js [hunk 3]
```diff
@@ -19,18 +19,17 @@ org.ekstep.contentrenderer.baseLauncher.extend({
     stageId:[],
     qid:[],
     enableHeartBeatEvent:false,
-
+    mimeType: ['application/vnd.ekstep.ecml-archive'],
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js]: Use Global constant object

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js [hunk 4]
```diff
@@ -19,18 +19,17 @@ org.ekstep.contentrenderer.baseLauncher.extend({
...
     /**
      * registers events
      * @memberof ecmlRenderer
      */
-    initLauncher: function(manifest) {
+    initLauncher: function() {
+        EkstepRendererAPI.dispatchEvent('renderer:launcher:register', this);
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js]: Use global events object

File: player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js [hunk 5]
```diff
@@ -13,10 +13,13 @@ org.ekstep.contentrenderer.baseLauncher.extend({
     lastPage: false,
     stageId:[],
     enableHeartBeatEvent: false,
+    mimeType: ['application/epub'],
     initLauncher: function () {
-        EkstepRendererAPI.addEventListener('content:load:application/vnd.ekstep.epub-archive', this.launch, this);
+        EkstepRendererAPI.dispatchEvent('renderer:launcher:register', this);
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js]: Clean & register should be in baseLaunher plugn itself. If any plugins wants to override clean, they can override by calling this.super().

File: player/public/coreplugins/org.ekstep.launcher-1.0/renderer/plugin.js [hunk 6]
```diff
@@ -8,73 +8,85 @@ Plugin.extend({
...
+            // this.loadCommonPlugins(() => {
+                content = contentObj;
+                // var contentTypePlugin = _.find(this.launcherMap, function(eachConfig) {
+                //     if (_.contains(eachConfig.mimeType, content.mimeType)) return eachConfig;
+                // });
             // Checking if mimetype launcher is already loaded or not
-            var pluginInstance = EkstepRendererAPI.getPluginObjs(contentTypePlugin.id);
-            if (pluginInstance) {
+                // var pluginInstance = EkstepRendererAPI.getPluginObjs(contentTypePlugin.manifest.id);
+                EkstepRendererAPI.dispatchEvent('renderer:launcher:clean')
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.launcher-1.0/renderer/plugin.js]: Look for callback & then do start of next content

**Pull request merged**